File: android/app/src/main/java/com/freelancekar/app/ui/screens/YouScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.Stripe
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@Composable
fun YouScreen(state: AppState, vm: AppViewModel) {
    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(horizontal = 20.dp).padding(top = 18.dp, bottom = 26.dp)) {
        Text("You", style = FkType.h1)
        Spacer(Modifier.height(18.dp))

        Row(verticalAlignment = Alignment.CenterVertically) {
            Stripe(Modifier.size(56.dp).clip(RoundedCornerShape(16.dp)).border(1.dp, FkColor.line09, RoundedCornerShape(16.dp)))
            Spacer(Modifier.width(13.dp))
            Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                Text("Café Mitti", style = FkType.screenTitle)
                Text("Client · ${cityOf(state)}", style = FkType.meta.copy(color = FkColor.text50))
            }
        }

        Spacer(Modifier.height(20.dp))
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            YouStat(state.bookings.size.toString(), "bookings", Modifier.weight(1f))
            YouStat(state.saved.size.toString(), "saved", Modifier.weight(1f))
            YouStat(toPayTotal(state), "to pay", Modifier.weight(1f))
        }

        Spacer(Modifier.height(20.dp))
        if (state.bookings.isEmpty()) {
            Column(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(15.dp))
                    .background(FkColor.card)
                    .border(1.dp, FkColor.line14, RoundedCornerShape(15.dp))
                    .padding(22.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                Text("No bookings yet. Hire someone and the order lives here.", style = FkType.bodySm.copy(color = FkColor.text45), textAlign = TextAlign.Center)
                Spacer(Modifier.height(14.dp))
                Box(
                    Modifier
                        .clip(RoundedCornerShape(11.dp))
                        .background(FkColor.lime)
                        .clickable { vm.go(Screen.EXPLORE) }
                        .padding(horizontal = 15.dp, vertical = 11.dp),
                ) {
                    Text("Find a freelancer", style = FkType.meta.copy(color = FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                }
            }
        } else {
            Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                state.bookings.forEach { b ->
                    Column(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(15.dp))
                            .background(FkColor.card)
                            .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp))
                            .clickable { vm.openOrderPublic(b.id) }
                            .padding(horizontal = 14.dp, vertical = 13.dp),
                    ) {
                        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                            Text(b.name, style = FkType.cardTitle)
                            Text(b.amount, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        }
                        Spacer(Modifier.height(4.dp))
                        Text(b.service, style = FkType.meta.copy(color = FkColor.text50))
                        Spacer(Modifier.height(10.dp))
                        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                            Text(
                                BOOKING_STATUS_LABEL[b.status] ?: "",
                                style = FkType.monoSm.copy(color = if (b.status == BookingStatus.APPROVED) FkColor.text45 else FkColor.lime),
                            )
                            Text(
                                if (b.status == BookingStatus.DELIVERED) "Approve →" else "View →",
                                style = FkType.meta.copy(color = FkColor.lime, fontWeight = androidx.compose.ui.text.font.FontWeight.Medium),
                            )
                        }
                    }
                }
            }
        }

        Spacer(Modifier.height(20.dp))
        Column(
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(15.dp))
                .background(FkColor.card)
                .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp)),
        ) {
            Row(
                Modifier
                    .fillMaxWidth()
                    .border(0.dp, FkColor.line07)
                    .clickable { vm.startFreelancer() }
                    .padding(14.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Text("Switch to freelancer mode", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.Medium))
                Text("→", style = FkType.meta.copy(color = FkColor.lime))
            }
            Box(Modifier.fillMaxWidth().height(1.dp).background(FkColor.line07))
            Text(
                "Payment methods",
                style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.Medium, color = FkColor.text.copy(alpha = 0.75f)),
                modifier = Modifier.fillMaxWidth().padding(14.dp),
            )
            Box(Modifier.fillMaxWidth().height(1.dp).background(FkColor.line07))
            Row(
                Modifier
                    .fillMaxWidth()
                    .clickable { vm.openHelp() }
                    .padding(14.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Text("Help & disputes", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.Medium))
                Text("→", style = FkType.meta.copy(color = FkColor.lime))
            }
        }
    }
}

@Composable
private fun YouStat(value: String, label: String, modifier: Modifier = Modifier) {
    Column(
        modifier
            .clip(RoundedCornerShape(14.dp))
            .background(FkColor.card)
            .border(1.dp, FkColor.line08, RoundedCornerShape(14.dp))
            .padding(horizontal = 6.dp, vertical = 13.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Text(value, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(16f, androidx.compose.ui.unit.TextUnitType.Sp)))
        Spacer(Modifier.height(3.dp))
        Text(label, style = FkType.metaXs.copy(color = FkColor.text45))
    }
}

File: ios/FreelanceKar/Views/Screens/YouScreen.swift
import SwiftUI

struct YouScreen: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                Text("You").font(.fkH1).padding(.bottom, 18)

                HStack(spacing: 13) {
                    Stripe(cornerRadius: 16).frame(width: 56, height: 56)
                        .overlay(RoundedRectangle(cornerRadius: 16).stroke(FK.line09, lineWidth: 1))
                    VStack(alignment: .leading, spacing: 3) {
                        Text("Café Mitti").font(.fkScreenTitle)
                        Text("Client · \(Selectors.cityOf(s))").font(.system(size: 12)).foregroundStyle(FK.text50)
                    }
                }

                HStack(spacing: 8) {
                    StatBox(value: String(s.bookings.count), label: "bookings")
                    StatBox(value: String(s.saved.count), label: "saved")
                    StatBox(value: Selectors.toPayTotal(s), label: "to pay")
                }
                .padding(.top, 20)

                VStack(spacing: 9) {
                    if s.bookings.isEmpty {
                        VStack(spacing: 14) {
                            Text("No bookings yet. Hire someone and the order lives here.")
                                .font(.fkBodySm).foregroundStyle(FK.text45).multilineTextAlignment(.center)
                            Button { store.go(.explore) } label: {
                                Text("Find a freelancer").font(.system(size: 12.5, weight: .semibold)).foregroundStyle(FK.onLime)
                                    .padding(.horizontal, 15).padding(.vertical, 11)
                            }
                            .background(FK.lime)
                            .clipShape(RoundedRectangle(cornerRadius: 11, style: .continuous))
                        }
                        .padding(22)
                        .frame(maxWidth: .infinity)
                        .background(FK.card)
                        .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 15).stroke(style: StrokeStyle(lineWidth: 1, dash: [5, 4])).foregroundStyle(FK.line14))
                    } else {
                        ForEach(s.bookings) { b in
                            Button { store.openOrder(b.id) } label: {
                                VStack(alignment: .leading, spacing: 0) {
                                    HStack {
                                        Text(b.name).font(.fkCardTitle)
                                        Spacer()
                                        Text(b.amount).font(.system(size: 13, weight: .semibold))
                                    }
                                    Text(b.service).font(.system(size: 11.5)).foregroundStyle(FK.text50).padding(.top, 1)
                                    HStack {
                                        Text(Selectors.bookingStatusLabel[b.status] ?? "")
                                            .font(.fkMonoSm)
                                            .foregroundStyle(b.status == .approved ? FK.text45 : FK.lime)
                                        Spacer()
                                        Text(b.status == .delivered ? "Approve →" : "View →")
                                            .font(.system(size: 11, weight: .medium)).foregroundStyle(FK.lime)
                                    }
                                    .padding(.top, 10)
                                }
                                .padding(.horizontal, 14).padding(.vertical, 13)
                            }
                            .foregroundStyle(FK.text)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                        }
                    }
                }
                .padding(.top, 20)

                VStack(spacing: 0) {
                    Button { store.startFreelancer() } label: {
                        HStack {
                            Text("Switch to freelancer mode").font(.system(size: 13, weight: .medium)).foregroundStyle(FK.text)
                            Spacer()
                            Text("→").font(.system(size: 12)).foregroundStyle(FK.lime)
                        }
                        .padding(14)
                    }
                    Divider().overlay(FK.line07)
                    HStack {
                        Text("Payment methods").font(.system(size: 13, weight: .medium)).foregroundStyle(Color(hex: 0xF2F2F0, opacity: 0.75))
                        Spacer()
                    }.padding(14)
                    Divider().overlay(FK.line07)
                    Button { store.openHelp() } label: {
                        HStack {
                            Text("Help & disputes").font(.system(size: 13, weight: .medium)).foregroundStyle(FK.text)
                            Spacer()
                            Text("→").font(.system(size: 12)).foregroundStyle(FK.lime)
                        }
                        .padding(14)
                    }
                }
                .background(FK.card)
                .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                .padding(.top, 20)
            }
            .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 26)
        }
    }
}

struct StatBox: View {
    var value: String
    var label: String
    var body: some View {
        VStack(alignment: .center, spacing: 3) {
            Text(value).font(.system(size: 16, weight: .semibold))
            Text(label).font(.system(size: 10)).foregroundStyle(FK.text45)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 13)
        .background(FK.card)
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line08, lineWidth: 1))
    }
}
